service/matching/task_queue_manager.go

```diff
@@ -827,9 +847,20 @@ func (c *taskQueueManagerImpl) fetchUserData(ctx context.Context) error {
 			TaskQueue:                fetchSource,
 			TaskQueueType:            enumspb.TASK_QUEUE_TYPE_WORKFLOW,
 			LastKnownUserDataVersion: knownUserData.GetVersion(),
```

Comment:
Since we got rid of `return err` on line 819, this looks like it would trigger an NPE. It actually returns 0 because proto getters explicitly handle nil receivers, but this looks very suspicious.

Reply:
Yeah I think it's not clear at first why it works, so probably better to just do `version = 0` and then `version = knownUserData.GetVersion()` if the error is nil.

Reply:
We call getters on possibly-nil proto messages all over the place. And also note that knownUserData can be nil even if err is nil